airbyte-integrations/connectors/source-close-com/source_close_com/source.py

```diff
@@ -69,6 +68,21 @@ def request_params(
 
         return params
 
+    def get_custom_field_schema(self) -> Mapping[str, Any]:
+        """Get custom field schema if it exists."""
+        if self.path() not in {"lead", "contact", "opportunity", "activity"}:
+            return {}
+        resp = requests.request("GET", url=f"{self.url_base}/custom_field/{self.path()}/", headers=self.authenticator.get_auth_header())
+        resp.raise_for_status()
+        resp_json: Mapping[str, Any] = resp.json()["data"]
+        return {f"custom.{data['id']}": {"type": ["null", "string", "number", "boolean"]} for data in resp_json}
+
+    def get_json_schema(self):
+        """Override default get_json_schema method to add custom fields to schema."""
+        schema = super().get_json_schema()
+        schema["properties"].update(self.get_custom_field_schema())
+        return schema
```

Comment:
I think it would be better to create a child class called `CloseComStreamCustomFields` and override the functions and change only to the streams it needs to use.

Reply:
Agreed - I updated to use a `CloseComStreamCustomFields` child class.